File: app/src/main/java/com/lkms/data/model/java/LogEntry.java
package com.lkms.data.model.java;

import com.google.gson.annotations.SerializedName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Đại diện cho một hàng trong bảng "LogEntry".
 */
@Data
@NoArgsConstructor
@AllArgsConstructor
public class LogEntry {

    @SerializedName("logId")
    private Integer logId;

    // Khóa ngoại đến ExperimentStep
    @SerializedName("experimentStepId")
    private Integer experimentStepId;

    @SerializedName("logType")
    private String logType;

    // Khóa ngoại đến User
    @SerializedName("userId")
    private Integer userId;

    @SerializedName("content")
    private String content;

    @SerializedName("url")
    private String url;

    @SerializedName("logTime")
    private String logTime;
}

File: app/src/main/java/com/lkms/data/model/LogEntry.kt
// Trong file: data/model/LogEntry.kt
package com.lkms.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Đại diện cho một hàng trong bảng "LogEntry".
 */
@Serializable
data class LogEntry(
    @SerialName("logId")
    val logId: Int? = null,

    // Khóa ngoại đến ExperimentStep
    @SerialName("experimentStepId")
    val experimentStepId: Int? = null,

    @SerialName("logType")
    val logType: String? = null,

    // Khóa ngoại đến User
    @SerialName("userId")
    val userId: Int? = null,

    @SerialName("content")
    val content: String? = null,

    @SerialName("url")
    val url: String? = null,

    @SerialName("logTime")
    val logTime: String? = null
)